packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -89,39 +99,213 @@ export class GitClientService {
         owner,
         repositoryName,
         branchName,
       });
-      if (!existingPullRequest) {
-        return this.provider.createPullRequestForBranch({
+
+      let pullRequest: PullRequest | undefined = existingPullRequest;
+
+      if (!pullRequest) {
+        pullRequest = await this.provider.createPullRequestForBranch({
           owner,
           repositoryName,
-          pullRequestTitle,
-          pullRequestBody,
+          pullRequestTitle: accumulativePullRequestTitle,
+          pullRequestBody: accumulativePullRequestBody,
           branchName,
           defaultBranchName: defaultBranch,
         });
       }
-      return existingPullRequest.url;
+
+      await this.provider.commentOnPullRequest({
+        where: { issueNumber: pullRequest.number, owner, repositoryName },
+        data: { body: pullRequestBody },
+      });
+
+      return pullRequest.url;
+    }
+
+    throw new InvalidPullRequestMode();
+  }
+
+  private async preCommitProcess({
+    gitClient,
+    branchName,
+    owner,
+    repositoryName,
+  }: PreCommitProcessArgs): PreCommitProcessResult {
+    console.log("Pre commit process");
+    await gitClient.git.checkout(branchName);
+
+    const commitsList = await this.provider.getCurrentUserCommitList({
+      branchName,
+      owner,
+      repositoryName,
+    });
+
+    const latestCommit = commitsList[0];
+
+    if (!latestCommit) {
+      throw new Error(
+        "Didn't find a commit that has been created by Amplication"
+      );
+    }
+
+    const { sha } = latestCommit;
+    const diff = await gitClient.git.diff([sha]);
+    if (diff.length === 0) {
+      console.log("Diff returned empty");
+      return { diff: null };
+    }
+    // Reset the branch to the latest commit
+    await gitClient.git.reset([sha]);
+    await gitClient.git.push(["--force"]);
+    await gitClient.resetState();
+    console.log("Diff returned");
+    return { diff };
+  }
+
+  async postCommitProcess({ diffPath, gitClient }: PostCommitProcessArgs) {
+    await gitClient.git.pull();
+
+    await gitClient.git
+      .applyPatch(diffPath, ["--3way", "--whitespace=nowarn"])
+      .add(["."])
+      .commit("Amplication diff restoration", undefined, {
+        "--author": "Amplication diff info@amplication.com",
+      })
+      .push();
+  }
+
+  private async restoreAmplicationBranchIfNotExists(
+    args: CreateBranchIfNotExistsArgs
+  ): Promise<Branch> {
+    const { branchName, owner, repositoryName, gitClient } = args;
+    const branch = await this.provider.getBranch(args);
+    if (!branch) {
+      const { defaultBranch } = await this.provider.getRepository(args);
+      const firstCommitOnDefaultBranch =
+        await this.provider.getFirstCommitOnBranch({
+          owner,
+          repositoryName,
+          branchName: defaultBranch,
+        });
+      const branch = await this.provider.createBranch({
+        owner,
+        branchName,
```

Comment:
let's try to avoid using `const` on the same var name (even it lives in a different context). This will help code readablity